push/sender/src/test/java/org/jboss/aerogear/unifiedpush/test/archive/UnifiedPushArchiveBase.java

```diff
@@ -0,0 +1,190 @@
+     */
+    private static final ArchivePath PATH_SERVICE_PROVIDERS = ArchivePaths.create(PATH_CLASSES, "META-INF/services");
+
+    // -------------------------------------------------------------------------------------||
+    // Instance Members -------------------------------------------------------------------||
+    // -------------------------------------------------------------------------------------||
+
+    // -------------------------------------------------------------------------------------||
+    // Constructor ------------------------------------------------------------------------||
+    // -------------------------------------------------------------------------------------||
+
+    /**
+     * Create a new WebArchive with any type storage engine as backing.
+     *
+     * @param delegate
+     *            The storage backing.
+     */
+    public UnifiedPushArchiveBase(final Archive<?> delegate) {
+        super(UnifiedPushArchive.class, delegate);
+    }
+
+    // -------------------------------------------------------------------------------------||
+    // Required Implementations -----------------------------------------------------------||
+    // -------------------------------------------------------------------------------------||
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.jboss.shrinkwrap.impl.base.container.ContainerBase#getManifestPath()
+     */
+    @Override
+    protected ArchivePath getManifestPath() {
+        return PATH_MANIFEST;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.jboss.shrinkwrap.impl.base.container.ContainerBase#getClassesPath()
+     */
+    @Override
+    protected ArchivePath getClassesPath() {
+        return PATH_CLASSES;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.jboss.shrinkwrap.impl.base.container.ContainerBase#getResourcePath()
+     */
+    @Override
+    protected ArchivePath getResourcePath() {
+        return PATH_RESOURCE;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.jboss.shrinkwrap.impl.base.container.ContainerBase#getLibraryPath()
+     */
+    @Override
+    protected ArchivePath getLibraryPath() {
+        return PATH_LIBRARY;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.jboss.shrinkwrap.impl.base.container.WebContainerBase#getWebPath()
+     */
+    @Override
+    protected ArchivePath getWebPath() {
+        return PATH_WEB;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.jboss.shrinkwrap.impl.base.container.WebContainerBase#getWebInfPath()
+     */
+    @Override
+    protected ArchivePath getWebInfPath() {
+        return PATH_WEB_INF;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.jboss.shrinkwrap.impl.base.container.WebContainerBase#getWebInfPath()
+     */
+    @Override
+    protected ArchivePath getServiceProvidersPath() {
+        return PATH_SERVICE_PROVIDERS;
+    }
+
+    public UnifiedPushArchive withMockito() {
```

Comment:
let's remove `return null` methods from abstract class ;-) leave it for final impl